File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.c
int findMin(int* nums, int numsSize) {
    int l = 0;
    int m = (numsSize - 1) / 2;
    int r = numsSize - 1;
    int lv, mv, rv;

    while (1) {
        lv = nums[l];
        mv = nums[m];
        rv = nums[r];

        if (lv == mv) {
            if (mv > rv) return rv;
            else return mv;
        }

        if (mv < rv) r = m;
        if (rv < mv) l = m;
        m = (l + r) / 2;
    }
}

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: list[int]) -> int:
        n = len(nums)
        l, m, r = 0, (n - 1) // 2, n - 1

        while True:
            lv = nums[l]
            mv = nums[m]
            rv = nums[r]

            if lv == mv: return min(mv, rv)

            if mv < rv: r = m
            if rv < mv: l = m
            m = (l + r) // 2
